Automattic/wp-calypso
# Stats: Fix data quantity for bar chart and partial date periods

Related to #83498

## Proposed Changes

* updating logic extending the quantity of data points requested for the bar char for partial periods (e.g. picking a day in the middle of the week and using weekly frequency)
* implement feedback from #83500

## Testing Instructions



* go to the live branch and the Traffic page
* select a period in the date picker that is in the middle of a week (start date), apply the request and verify that the first visible bar includes the requested date
* select a period in the date picker that is in the middle of a month (start date), apply the request and verify that the first visible bar includes the requested date
* select a period in the date picker that is in the middle of a year (start date), apply the request and verify that the first visible bar includes the requested date

## Pre-merge Checklist
